numba/targets/arraymath.py

```diff
@@ -1423,6 +1437,18 @@ def np_cov(m, y=None, rowvar=True, bias=False, ddof=None):
         check_dimensions(m, 'm')
         check_dimensions(y, 'y')
 
+        # reject problem if ddof invalid (either upfront if type is
+        # obviously invalid, or later if value found to be non-integral)
+        if ddof in (None, types.none):
+            _DDOF_HANDLER = _handle_ddof_nop
+        else:
+            if isinstance(ddof, (types.Integer, types.Boolean)):
+                _DDOF_HANDLER = _handle_ddof_nop
+            elif isinstance(ddof, types.Float):
+                _DDOF_HANDLER = _handle_ddof
+            else:
+                raise TypeError('ddof must be integer')
```

Comment:
Think this can be a `TypingError` as it'll be caught at compile time? Does `ddof must be a numerical scalar type` make more sense too as it's a typing problem?

Reply:
Done